src/backend/distributed/operations/shard_split.c

```diff
@@ -0,0 +1,920 @@
+
+	ExecuteTaskListOutsideTransaction(ROW_MODIFY_NONE, splitCopyTaskList,
+									  MaxAdaptiveExecutorPoolSize,
+									  NULL /* jobIdList (ignored by API implementation) */);
+}
+
+
+/*
+ * Create Copy command for a given shard source shard to be copied to corresponding split children.
+ * 'sourceShardSplitInterval' : Source shard interval to be copied.
+ * 'splitChildrenShardINnerIntervalList' : List of shard intervals for split children.
+ * 'destinationWorkerNodesList' : List of workers for split children placement.
+ * Here is an example of a 2 way split copy :
+ * SELECT * from worker_split_copy(
+ *  81060000, -- source shard id to split copy
+ *  ARRAY[
+ *       -- split copy info for split children 1
+ *      ROW(81060015, -- destination shard id
+ *           -2147483648, -- split range begin
+ *          1073741823, --split range end
+ *          10 -- worker node id)::citus.split_copy_info,
+ *      -- split copy info for split children 2
+ *      ROW(81060016,  --destination shard id
+ *          1073741824, --split range begin
+ *          2147483647, --split range end
+ *          11 -- workef node id)::citus.split_copy_info
+ *      ]
+ *  );
+ */
+static StringInfo
+CreateSplitCopyCommand(ShardInterval *sourceShardSplitInterval,
+					   List *splitChildrenShardIntervalList,
+					   List *destinationWorkerNodesList)
+{
+	StringInfo splitCopyInfoArray = makeStringInfo();
+	appendStringInfo(splitCopyInfoArray, "ARRAY[");
+
+	ShardInterval *splitChildShardInterval = NULL;
+	bool addComma = false;
+	WorkerNode *destinationWorkerNode = NULL;
+	forboth_ptr(splitChildShardInterval, splitChildrenShardIntervalList,
+				destinationWorkerNode, destinationWorkerNodesList)
+	{
+		if (addComma)
+		{
+			appendStringInfo(splitCopyInfoArray, ",");
+		}
+
+		StringInfo splitCopyInfoRow = makeStringInfo();
+		appendStringInfo(splitCopyInfoRow,
+						 "ROW(%lu, %d, %d, %u)::citus.split_copy_info",
+						 splitChildShardInterval->shardId,
+						 DatumGetInt32(splitChildShardInterval->minValue),
+						 DatumGetInt32(splitChildShardInterval->maxValue),
+						 destinationWorkerNode->nodeId);
+		appendStringInfo(splitCopyInfoArray, "%s", splitCopyInfoRow->data);
+
+		addComma = true;
+	}
+	appendStringInfo(splitCopyInfoArray, "]");
+
+	StringInfo splitCopyUdf = makeStringInfo();
+	appendStringInfo(splitCopyUdf, "SELECT worker_split_copy(%lu, %s);",
+					 sourceShardSplitInterval->shardId,
+					 splitCopyInfoArray->data);
+
+	return splitCopyUdf;
+}
+
+
+/*
+ * Create an object on a worker node.
+ */
+static void
+CreateObjectOnPlacement(List *objectCreationCommandList,
+						WorkerNode *workerPlacementNode)
+{
+	char *currentUser = CurrentUserName();
+	SendCommandListToWorkerOutsideTransaction(workerPlacementNode->workerName,
+											  workerPlacementNode->workerPort,
+											  currentUser,
+											  objectCreationCommandList);
+}
+
+
+/*
+ * Create split children intervals for a shardgroup given list of split points.
+ */
```

Comment:
this comment should probably elaborate more on the data structure that's being returned

Reply:
Done.